### PR TITLE
[bug-fix] root_dir is not set in copyThroughBuffers

### PR DESCRIPTION
**Pull request by lingo-xp**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

root_dir is not set in copyThrouahBuffers function in DiskLocal.cpp. In this case, when calling DiskLocal::copyDIrectoryContent function, it will also copy parent dir name of from path, it is not expected.
ex. if we want to copy aa.cpp,bb.cpp from  /store1/xxx/xxx/111 to /store2/xxx/xxx/tmp_111, the final path will be /store2/xxx/xxx/tmp_111/111/{aa.cpp,bb.cpp}
the correct one shold be /store2/xxx/xxx/tmp_111/{aa.cpp,bb.cpp}

### CODE REVIEW

File: src/Disks/DiskLocal.cpp [hunk 1]
```diff
@@ -437,15 +437,15 @@ void DiskLocal::copy(const String & from_path, const std::shared_ptr<IDisk> & to
         fs::copy(from, to, fs::copy_options::recursive | fs::copy_options::overwrite_existing); /// Use more optimal way.
     }
     else
-        copyThroughBuffers(from_path, to_disk, to_path); /// Base implementation.
+        copyThroughBuffers(from_path, to_disk, to_path, true); /// Base implementation.
```
Comment (tavplubix) [inline src/Disks/DiskLocal.cpp:440]:
```suggestion
        copyThroughBuffers(from_path, to_disk, to_path, /* copy_root_dir */ true); /// Base implementation.
```

File: src/Disks/DiskLocal.cpp [hunk 2]
```diff
@@ -437,15 +437,15 @@ void DiskLocal::copy(const String & from_path, const std::shared_ptr<IDisk> & to
...
 }
 
 void DiskLocal::copyDirectoryContent(const String & from_dir, const std::shared_ptr<IDisk> & to_disk, const String & to_dir)
 {
     if (isSameDiskType(*this, *to_disk))
         fs::copy(from_dir, to_dir, fs::copy_options::recursive | fs::copy_options::overwrite_existing); /// Use more optimal way.
     else
-        copyThroughBuffers(from_dir, to_disk, to_dir); /// Base implementation.
+        copyThroughBuffers(from_dir, to_disk, to_dir, false); /// Base implementation.
```
Comment (tavplubix) [inline src/Disks/DiskLocal.cpp:448]:
```suggestion
        copyThroughBuffers(from_dir, to_disk, to_dir, /* copy_root_dir */ false); /// Base implementation.
```

**Review by tavplubix — APPROVED** [review]
Nice catch, thanks!

**Comment by tavplubix** [comment]
Could you please fix PR description? Changelog entry is required and it must be user-readable.

**Comment by lingo-xp** [comment]
sure,
Changelog entry:
Fix the issue that copyDirectoryContent function in DIskLocal.cpp copys files to wrong path

**Pull request merged**